Added channels model and removed user from Video model

giocalitri opened the pull request:
#### What are the relevant tickets?
closes #51 

#### What's this PR do?
Adds the channels model, rest APIs for them and changes the way the site can be navigated

#### How should this be manually tested?
Start the app, create a channel and upload videos in that channels

#### Any background context you want to provide?
This is not a backwards compatible migration. I suggest to delete all the videos from the app **BEFORE** applying this migration, or you just need to manually associate the orphan videos to a new channel from the admin.

mbertrand commented on ui/models.py:
```diff
@@ -20,6 +20,16 @@
+    title = models.TextField()
+    description = models.TextField(null=True, blank=True)
+    slug = models.UUIDField(unique=True, null=False, blank=False, default=uuid4)
+    owners = models.ManyToManyField(settings.AUTH_USER_MODEL)
```
Any advantage of this approach vs having one creator/owner and using moira_lists to determine who can administer/edit/view?

giocalitri replied:
Not really. The idea was to avoid to corner ourselves in a spot where we need to have different users that actually own the channel.
But the more I think about it, the less I am convinced.
I think I will move it to a foreignKey

giocalitri commented:
Ok. I think it is time to write some tests around.

mbertrand commented on ui/api_test.py:
```diff
@@ -1,122 +1,93 @@
+def test_process_dropbox_data_happy_path(mocker):
     """
-    Verify that Video.job_status property returns the status of its encoding job
+    Tests that the process_dropbox_data in case everething is fine
```
Minor type `everething`

mbertrand commented on ui/forms.py:
```diff
@@ -52,24 +52,38 @@ def to_python(self, values):
     """
     moira_lists = DynamicModelMultipleChoiceField(
-        model=MoiraList, field="name",
+        model=models.MoiraList,
```
Maybe `moira_lists` should be associated with collections instead of videos now?

mbertrand commented on ui/models.py:
```diff
@@ -20,6 +20,36 @@
+    title = models.TextField()
+    description = models.TextField(null=True, blank=True)
+    owner = models.ForeignKey(settings.AUTH_USER_MODEL)
+
```
Add `moira_lists` to `Collection` and remove from `Video`?

giocalitri commented:
@mbertrand I updated the models and fixed related problems with REST APIs and serializers.
I completely removed the moira lists from the UI for the time being.

mbertrand commented on ui/templates/ui/collection_list.html:
```diff
@@ -0,0 +1,63 @@
+
+<hr>
+
+<h2>Create New Collection</h2>
```
@giocalitri if you want to hold off on the form issue until the next PR for #118 that's fine with me.  Everything else works great.

giocalitri replied:
that is related to the fact that the current UI does not makes ajax requests with the right content type.
Anyway I put in place a quick fix and test

mbertrand commented:
👍
